private-templates-service/adaptivecards-templating-service/src/models/mongo/TemplateModel.ts

```diff
@@ -33,3 +38,11 @@ export const TemplateSchema: Schema = new Schema(
     timestamps: { createdAt: true, updatedAt: true }
   }
 );
+TemplateSchema.pre("validate", function(next) {
+  this._id = MongoUtils.generateUniqueID().toHexString();
+  next();
+});
+
+TemplateSchema.index({ createdAt: 1 });
```

Comment:
Don't quite understand these 3 lines - guessing they'll be changed later?

Reply:
These 3 lines create index for the keys which we will run sort on.